qsimcirq/qsim_simulator.py

```diff
@@ -718,3 +857,33 @@ def _translate_circuit(
             self._translated_circuits.append((circuit, translated_circuit))
 
         return translated_circuit
+
+    def _translate_moments(
+        self,
+        circuit: Any,
+        translator_fn_name: str,
+        qubit_order: cirq.QubitOrderOrList,
+        is_noisy: bool,
+    ):
+        """Translates a circuit by moments.
+
+        Returns:
+            A tuple containing the translated circuit and the gate indices
+            corresponding to moment boundaries.
+        """
+        reindex = [0]
+        full_circuit = qsim.NoisyCircuit() if is_noisy else qsim.Circuit()
```

Comment:
I would probably do this in a different way. I would modify `translate_cirq_to_qsim` and  `translate_cirq_to_qtrajectory` in qsim_circuit.py to return the gate indices corresponding to moment boundaries in addition to returning translated circuits. In this case, `qsim.compose_noisy_circuits` and `qsim.compose_circuits` are not required.

Reply:
Sounds good to me, let's see how the CI tests feel about it.